proposals/Fleet-Windows-OS-vulns.md

```diff
@@ -0,0 +1,139 @@
+basically we have three problems we need to solve: 
+1. We need a list of vulnerabilities + remediations. Each remediation is composed of a patch + some steps.
+2. For a given vulnerability we need to determine whether the proper patch was applied.
+3. For a given vulnerability we need to determine whether the steps (if any) were followed.
+
+## List of vulnerabilities
+
+To get the list of vulnerabilities we can use the [Microsoft Security Updates API](https://api.msrc.microsoft.com/cvrf/v2.0/swagger/index). This [endpoint](api.msrc.microsoft.com/cvrf/v2.0/document/yyyy-mm) in particular exposes all the security updates released for yyyy-mmm formatted according to the [Common Vulnerability Reporting Framework](http://docs.oasis-open.org/csaf/csaf-cvrf/v1.2/csaf-cvrf-v1.2.html#:~:text=The%20CSAF%20Common%20Vulnerability%20Reporting,impact%20and%20remediation%20among%20interested) (cvrf) format, this cvrf document will include entries for vulnerabilities, how to patch them and what products are affected. 
+
+For example, say we have a host running *Windows 10 Version 21H2 x64* and we want to determine whether that host is susceptible to *CVE-2022-26925*. Looking at https://api.msrc.microsoft.com/cvrf/v2.0/document/2022-May we can see that  *Windows 10 Version 21H2 for x64* is assigned the Product ID **11931**:
+
+```
+...
+<prod:ProductTree>
+    ...
+    <prod:Branch Type="Product Family" Name="Windows">
+    ...
+        <prod:FullProductName ProductID="11931">Windows 10 Version 21H2 for x64-based Systems</prod:FullProductName>
+        ...
+    </prod:Branch>
+    …
+<prod:ProductTree>
+…
+```
+
+Looking at the matching vulnerability element (`<vuln:Vulnerability Ordinal="47">`) *CVE-2022-26925*, we can see that
+*Windows 10 Version 21H2 x64* is indeed affected by it:
+
+```
+<vuln:ProductStatuses>
+    <vuln:Status Type="Known Affected">
+    ...
+        <vuln:ProductID>11923</vuln:ProductID>
+    ...
+    </vuln:Status>
+</vuln:ProductStatuses>
+```
+
+And also the remediation steps:
+
+```
+<vuln:Remediations>
+…
+    <vuln:Remediation Type="Vendor Fix">
+        <vuln:Description>5013942</vuln:Description>
+        <vuln:URL>https://catalog.update.microsoft.com/v7/site/Search.aspx?q=KB5013942</vuln:URL>
+        <vuln:Supercedence>5012599</vuln:Supercedence>
+        ..
+        <vuln:ProductID>11931</vuln:ProductID>
+        ...
+        <vuln:RestartRequired>Yes</vuln:RestartRequired>
+        …
+    </vuln:Remediation>
+    <vuln:Remediation Type="Known Issue">
+        <vuln:Description>5013942</vuln:Description>
+        <vuln:URL>https://support.microsoft.com/help/5013942</vuln:URL>
+        ...
+        <vuln:ProductID>11931</vuln:ProductID>
+    </vuln:Remediation>
+</vuln:Remediations>
+```
+
+For the previous extract we can determine four things:
+1. The vulnerability was patched (‘Vendor Fix’) on the **KB5013942** update.
+2. **KB5013942** supersedes **KB5012599** (this is useful for handling cumulative updates).
```

Comment:
What does it mean to supersede? Can you provide vuln KB5012599?

Reply:
supersede means to replace, in context of cumulative updates, if we say **KB5013942** supersedes **KB5012599** it means that **KB5013942** contains all the patches included in **KB5012599** plus new patches.

> Can you provide vuln KB5012599?

**KB5012599** was used to patch CVE-2022-22008